server/re/src/irods_re_serialization.cpp

```diff
@@ -1199,6 +1199,77 @@ namespace irods::re_serialization
         }
     } // serialize_vector_of_strings_ptr
 
+    static irods::error serialize_execMyRuleInp_ptr(boost::any _p, serialized_parameter_t& _out)
+    {
+        try {
+            log_re::trace(__func__);
+            const auto* casted_ptr = boost::any_cast<execMyRuleInp_t*>(_p);
+            if(!casted_ptr) {
+                _out["execMyRuleInp_ptr"] = "nullptr";
+                return SUCCESS();
+            }
+            _out["myRule"] = casted_ptr->myRule;
+            _out["addr_hostAddr"] = casted_ptr->addr.hostAddr;
+            _out["addr_zoneName"] = casted_ptr->addr.zoneName;
+            _out["addr_portNum"] = boost::lexical_cast<std::string>(casted_ptr->addr.portNum);
+            _out["addr_dummyInt"] = boost::lexical_cast<std::string>(casted_ptr->addr.dummyInt);
+            serialize_keyValPair(casted_ptr->condInput, _out);
+            _out["outParamDesc"] = casted_ptr->outParamDesc;
+
+            const auto* msparamary_ptr = casted_ptr->inpParamArray;
+            if(!msparamary_ptr) {
+                _out["inpParamArray"] = "nullptr";
+                return SUCCESS();
+            }
+            _out["inpParamArray_len"] = boost::lexical_cast<std::string>(msparamary_ptr->len);
+            _out["inpParamArray_oprType"] = boost::lexical_cast<std::string>(msparamary_ptr->oprType);
+            for (int i = 0; i < msparamary_ptr->len; i++) {
+                std::string prefix = "inpParamArray_" + std::to_string(i) + "_";
+                _out[prefix + "label"] = msparamary_ptr->msParam[i]->label;
+                _out[prefix + "type"] = msparamary_ptr->msParam[i]->type;
+                const auto* msparam_ptr = msparamary_ptr->msParam[i];
+
+                if (msparam_ptr == NULL || msparam_ptr->inOutStruct == NULL) {
+                    _out[prefix + "inOutStruct"] = "null";
+                }
+                else if (strcmp(msparam_ptr->type, STR_MS_T) != 0) {
+                    _out[prefix + "inOutStruct"] = "non_string_value";
+                }
+                else {
+                    _out[prefix + "inOutStruct"] = static_cast<char*>(msparam_ptr->inOutStruct);
+                }
+                const auto* ptr = msparamary_ptr->msParam[i]->inpOutBuf;
+                if (ptr) {
+                    _out[prefix + "inpOutBuf_len"] = boost::lexical_cast<std::string>(ptr->len);
+                    std::string& str = _out[prefix + "inpOutBuf_buf"];
+
+                    for (int j = 0; j < ptr->len; j++) {
+                        unsigned char c = *((unsigned char*) (ptr->buf) + j);
+
+                        if (isprint(c) || isspace(c) || isblank(c)) {
+                            str += c;
+                        }
+                        else {
+                            str += fmt::format("\\x{0:02x}", c);
+                        }
+                    }
+                }
+                else {
+                    _out[prefix + "null_value"] = "null_value";
```

Comment:
Can this result in overwriting previous mappings?

Also, `= "null_value"` -> `= "null"`?

Reply:
I do not believe any previous key is named `null_value`, so I don't think a collision is possible here.
As for the assigned value, this block was mostly lifted from the `bytesBuf_t` serializer elsewhere in this file. I thought to keep it the same to be consistent, but I can change this accordingly if you think it would be better otherwise.

Reply:
That's fine. Leave it as is.